.github/workflows/centos_7.yml

```diff
@@ -54,18 +54,33 @@ jobs:
           submodules: recursive
       - uses: ./.github/actions/environment
       - name: packaging
+        id: packaging
         env:
           RWS_AUTH: ${{ secrets.RWS_AUTH }}
           OS: 'el'
           DIST: '7'
           GC64: ${{ matrix.build-type == 'gc64' }}
         uses: ./.github/actions/pack_and_deploy
-      - name: call action to send Telegram message on failure
-        env:
-          TELEGRAM_TOKEN: ${{ secrets.TELEGRAM_CORE_TOKEN }}
-          TELEGRAM_TO: ${{ secrets.TELEGRAM_CORE_TO }}
-        uses: ./.github/actions/send-telegram-notify
+      - name: Set chat for reports
```

Comment:
It looks like your approach has some downsides and the main one is code duplication. What about creating an internal composite action, say `report-job-status`, and moving this logic to it? Look at this:

```
- name: Send VK Teams message on failure
  if: failure()
  uses: ./.github/actions/report-job-status
  with:
    job-steps: ${{ ToJson(steps) }}
  env:
    VKTEAMS_API_URL: ${{ secrets.VKTEAMS_API_URL }}
    VKTEAMS_BOT_TOKEN: ${{ secrets.VKTEAMS_BOT_TOKEN }}
    VKTEAMS_CHAT_ID: ${{ secrets.VKTEAMS_CHAT_ID }}
```

or

```
- name: Send VK Teams message on failure
  if: failure()
  uses: ./.github/actions/report-job-status
  with:
    api-url: ${{ secrets.VKTEAMS_API_URL }}
    bot-token: ${{ secrets.VKTEAMS_BOT_TOKEN }}
    chat-id: ${{ secrets.VKTEAMS_CHAT_ID }}
    job-steps: ${{ ToJson(steps) }}
```

Reply:
Done